pkg/frontend/querymiddleware/split_by_instant_interval.go

```diff
@@ -175,3 +177,20 @@ func (s *splitInstantQueryByIntervalMiddleware) Do(ctx context.Context, req Requ
 		Headers: shardedQueryable.getResponseHeaders(),
 	}, nil
 }
+
+// getSplitIntervalForQuery calculates and return the split interval that should be used to run the instant query.
+func (s *splitInstantQueryByIntervalMiddleware) getSplitIntervalForQuery(tenantsIds []string, r Request, spanLog log.Logger) time.Duration {
+	// Check if splitting is disabled for the given request.
+	if r.GetOptions().InstantSplitDisabled {
+		return 0
+	}
+
```

Comment:
Shouldn't we honor `req.InstantSplitInterval()` if set?

Reply:
Yes 😅 completely missed the most important point of the PR. Good catch 👀